nexus/src/app/switch_port.rs

```diff
@@ -289,6 +296,48 @@ impl super::Nexus {
 
         Ok(())
     }
+
+    pub(crate) async fn switch_port_status(
+        &self,
+        opctx: &OpContext,
+        switch: Name,
+        port: Name,
```

Comment:
Is `Name` really the right type here? I'd have expected this to take `SwitchLocation` directly.

Reply:
There was a lot of back and fort on this with the switch APIs first landed. And `Name` is where things landed, I cannot recall exactly why. This is an effort to be consistent with what's already there.